src/nemos/observation_models.py

```diff
@@ -797,9 +797,7 @@ def deviance(
         where $ y $ is the observed data, $ \hat{y} $ is the predicted data, and $\text{LL}$ is the model
         log-likelihood. Lower values of deviance indicate a better fit.
         """
-        y_mu = jnp.clip(
-            neural_activity / predicted_rate, min=jnp.finfo(predicted_rate.dtype).eps
-        )
+        y_mu = jnp.clip(neural_activity / predicted_rate, min=jnp.finfo(float).eps)
```

Comment:
@billbrod this matches statsmodels computation. it is more precise than what we had previously (which for jax is usually float32), and resulted in discrepancies with statsmodels due to very low values.